lab_6_pipeline/pipeline.py

```diff
@@ -27,16 +41,46 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
         Args:
             path_to_raw_txt_data (pathlib.Path): Path to raw txt data
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validate folder with assets.
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if not any(self.path_to_raw_txt_data.iterdir()):
+            raise EmptyDirectoryError
+
+        raw_files = list(self.path_to_raw_txt_data.glob("*_raw.txt"))
+        meta_files = list(self.path_to_raw_txt_data.glob("*_meta.json"))
+        if len(raw_files) != len(meta_files):
+            raise InconsistentDatasetError
+
+        sorted_raw = sorted(raw_files)
+        sorted_meta = sorted(meta_files)
+
+        for ind, (raw, meta) in enumerate(zip(sorted_raw, sorted_meta), 1):
+            if ind != get_article_id_from_filepath(raw) or\
+                ind != get_article_id_from_filepath(meta) or\
+                raw.stat().st_size == 0 or meta.stat().st_size == 0:
+                raise InconsistentDatasetError
 
     def _scan_dataset(self) -> None:
         """
         Register each dataset entry.
         """
+        for file in list(self.path_to_raw_txt_data.glob("*_raw.txt")):
+            i = get_article_id_from_filepath(file)
```

Comment:
Consider using a more meaningful and pylint compliant variable name

Reply:
ok, thank you